tests/modules/helloscripting.c

```diff
@@ -0,0 +1,354 @@
+#include "valkeymodule.h"
+
+#include <ctype.h>
+#include <errno.h>
+#include <string.h>
+
+/*
+ * This module implements a very simple stack based scripting language.
+ * It's purpose is only to test the valkey module API to implement scripting
+ * engines.
+ *
+ * The language is called HELLO, and a program in this language is formed by
+ * a list of function definitions.
+ * The language only supports 32-bit integer, and it only allows to return an
+ * integer constant, or return the value passed as the first argument to the
+ * function.
+ *
+ * Example of a program:
+ *
+ * ```
+ * FUNCTION foo
+ * ARGS 0        # pushes the value in the first argument to the top of the
+ * stack RETURN        # returns the current value on the top of the stack
```

Comment:
```suggestion
 * ARGS 0        # pushes the value in the first argument to the top of the
 *               # stack
 * RETURN        # returns the current value on the top of the stack
```

Not mentioned:

* FUNCTION starts a function?
* RETURN also marks the end of the function?